### PR TITLE
Address the output option and add unit tests

### PR DESCRIPTION
**Pull request by Akrion**
Fixes the output option and adds unit tests for it

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Running your Dangerfile has Failed</td>
    </tr>
  </tbody>
</table>




## Danger has errored

Error: TypeError

```sh
TypeError: Cannot read property 'includes' of null
    at Object.<anonymous> (dangerfile.js:28:40)
    at Runtime._execModule (/home/ubuntu/directory-metagen/node_modules/jest-runtime/build/index.js:447:13)
    at Runtime.requireModule (/home/ubuntu/directory-metagen/node_modules/jest-runtime/build/index.js:295:14)
    at /home/ubuntu/directory-metagen/node_modules/danger/distribution/runner/DangerfileRunner.js:136:33
    at ensureCleanDangerfile (/home/ubuntu/directory-metagen/node_modules/danger/distribution/runner/DangerfileRunner.js:176:5)
    at Object.<anonymous> (/home/ubuntu/directory-metagen/node_modules/danger/distribution/runner/DangerfileRunner.js:135:21)
    at step (/home/ubuntu/directory-metagen/node_modules/danger/distribution/runner/DangerfileRunner.js:32:23)
    at Object.next (/home/ubuntu/directory-metagen/node_modules/danger/distribution/runner/DangerfileRunner.js:13:53)
    at /home/ubuntu/directory-metagen/node_modules/danger/distribution/runner/DangerfileRunner.js:7:71
    at __awaiter (/home/ubuntu/directory-metagen/node_modules/danger/distribution/runner/DangerfileRunner.js:3:12)
```


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: index.js [hunk 1]
```diff
@@ -15,9 +16,12 @@ let test = regex => str => regex.test(str) // This mirrors the test function in
...
 // Core
 let defaultFilter = _.filter(test(/.js|.html|.jsx|.ts|.coffee|.less|.css|.sass|.hbs|.ejs/))
-let metagen = dir => relativeFilenames(dir.path, dir.exclusions || [dir.output || '__all.js'])
-  .then(dir.filter || defaultFilter)
-  .then(files => fs.writeFileAsync(dir.path + (dir.output || '__all.js'), dir.format(files, dir)))
+let metagen = options => relativeFilenames(options.path, options.exclusions || [options.output || '__all.js'])
+  .then(options.filter || defaultFilter)
+  .then(files => fs.writeFileAsync(
+      path.join(options.path, (options.output || '__all.js')),
+      options.format(files.map(file => file.replace(/^\//g, '')), options))
```
Comment (daedalus28) [inline index.js]: we can use lodash/fp's `replace` so we don't need to create a function inside map and leverage the curry

File: index.js [hunk 2]
```diff
@@ -15,9 +16,12 @@ let test = regex => str => regex.test(str) // This mirrors the test function in
 
 // Core
 let defaultFilter = _.filter(test(/.js|.html|.jsx|.ts|.coffee|.less|.css|.sass|.hbs|.ejs/))
-let metagen = dir => relativeFilenames(dir.path, dir.exclusions || [dir.output || '__all.js'])
-  .then(dir.filter || defaultFilter)
-  .then(files => fs.writeFileAsync(dir.path + (dir.output || '__all.js'), dir.format(files, dir)))
+let metagen = options => relativeFilenames(options.path, options.exclusions || [options.output || '__all.js'])
+  .then(options.filter || defaultFilter)
+  .then(files => fs.writeFileAsync(
+      path.join(options.path, (options.output || '__all.js')),
```
Comment (daedalus28) [inline index.js]: we don't need to wrap the output in parenthesis

File: index.js (hunk 1 above)
Comment (daedalus28) [inline index.js]: additionally, we probably want to pull out the pending `_.replace(/^\//g, '')` to the path utils and give a name like `stripLeadingSlash`

Comment (daedalus28) [inline index.js]: Just for traceability, this feedback is tied to these two entries from the contributor guide:

`Use lodash/fp`
`Create temporary variables... [if they] require a comment documenting what they do`

**Pull request merged**